crates/solvers/src/boundary/baseline.rs

```diff
@@ -44,47 +44,40 @@ impl<'a> Solver<'a> {
             max_hops,
         );
 
-        let segments = match request.side {
-            order::Side::Buy => {
-                let (segments, _) = candidates
-                    .iter()
-                    .filter_map(|path| {
-                        let estimate = baseline_solver::estimate_sell_amount(
-                            request.buy.amount,
-                            path,
-                            &self.amms,
-                        )?;
-                        let segments = self.traverse_path(
-                            &estimate.path,
-                            request.sell.token.0,
-                            estimate.value,
-                        )?;
-
-                        let sell = estimate.value;
-                        let buy = segments
-                            .last()
-                            .map(|segment| segment.output.amount)
-                            .unwrap_or(sell);
-
-                        if sell <= request.sell.amount && buy >= request.buy.amount {
-                            Some((segments, sell))
-                        } else {
-                            None
-                        }
-                    })
-                    .min_by_key(|(_, sell)| *sell)?;
-                segments
-            }
-            order::Side::Sell => {
-                let estimate = candidates
-                    .iter()
-                    .filter_map(|path| {
-                        baseline_solver::estimate_buy_amount(request.sell.amount, path, &self.amms)
-                    })
-                    .filter(|estimate| estimate.value >= request.buy.amount)
-                    .max_by_key(|estimate| estimate.value)?;
-                self.traverse_path(&estimate.path, request.sell.token.0, request.sell.amount)?
-            }
+        let (segments, _) = match request.side {
+            order::Side::Buy => candidates
+                .iter()
+                .filter_map(|path| {
+                    let sell = baseline_solver::estimate_sell_amount(
+                        request.buy.amount,
+                        path,
+                        &self.amms,
+                    )?;
+                    let segments =
+                        self.traverse_path(&sell.path, request.sell.token.0, sell.value)?;
+                    let buy = segments
+                        .last()
+                        .map(|segment| segment.output.amount)
+                        .unwrap_or(sell.value);
+
+                    (sell.value <= request.sell.amount && buy >= request.buy.amount)
```

Comment:
We are doing the two sided check for buy order but not for sell orders. Should we add it for sell orders as well to be safe? Also, I'm wondering if we should log in the unexpected case that `buy < request.buy.amount`

Reply:
> We are doing the two sided check for buy order but not for sell orders.

Mostly because we don't have the same round-tripping logic for sell orders, so it can't ever happen. But I agree that we should likely do a check for both (as there is an implicit assumption on the behavior of the `baseline_solver` module code).

> Also, I'm wondering if we should log in the unexpected case that `buy < request.buy.amount`

Will add for both sell and buy orders.